File: 2026-03-March-GFG-POTD/March 03 - Longest subarray with Atmost two distinct integers/Solution.py
class Solution:
    def totalElements(self, arr):
        left = 0
        max_length = 0
        freq = {}
        
        for right in range(len(arr)):
            # Add current element
            freq[arr[right]] = freq.get(arr[right], 0) + 1
            
            # Shrink window if more than 2 distinct elements
            while len(freq) > 2:
                freq[arr[left]] -= 1
                
                if freq[arr[left]] == 0:
                    del freq[arr[left]]
                
                left += 1
            
            # Update maximum length
            max_length = max(max_length, right - left + 1)
        
        return max_length

File: 2026-03-March-GFG-POTD/March 03 - Longest subarray with Atmost two distinct integers/Solution.cpp
#include <unordered_map>
#include <vector>
using namespace std;

class Solution {
public:
    int totalElements(vector<int>& arr) {
        int left = 0;
        int maxLength = 0;
        unordered_map<int, int> freq;
        
        for (int right = 0; right < arr.size(); right++) {
            // Add current element
            freq[arr[right]]++;
            
            // Shrink window if more than 2 distinct elements
            while (freq.size() > 2) {
                freq[arr[left]]--;
                
                if (freq[arr[left]] == 0) {
                    freq.erase(arr[left]);
                }
                
                left++;
            }
            
            // Update maximum length
            maxLength = max(maxLength, right - left + 1);
        }
        
        return maxLength;
    }
};
